File: src/main/java/micronaut/playground/web/HelloController.java
package micronaut.playground.web;

import io.micronaut.http.annotation.Controller;
import io.micronaut.http.annotation.Get;
import io.reactivex.Single;
import micronaut.playground.client.model.Hello;
import micronaut.playground.domain.service.Greeter;

import javax.inject.Inject;

@Controller
public class HelloController {

  private Greeter greeter;

  @Inject
  public HelloController(Greeter greeter) {
    this.greeter = greeter;
  }

  @Get("/{name}")
  public Single<Hello> hello(String name) {
    return greeter.greetHello(name);
  }
}

File: src/main/kotlin/micronaut/playground/web/HelloController.kt
package micronaut.playground.web

import io.micronaut.http.MediaType
import io.micronaut.http.annotation.Controller
import io.micronaut.http.annotation.Get
import io.micronaut.http.annotation.Produces
import io.micronaut.scheduling.TaskExecutors
import io.micronaut.scheduling.annotation.ExecuteOn
import micronaut.playground.client.model.Hello
import micronaut.playground.domain.service.Greeter
import org.reactivestreams.Publisher

@Controller("/api/hello")
class HelloController(val greeter: Greeter) {

    @Get("/{name}")
    @ExecuteOn(TaskExecutors.IO)
    @Produces(MediaType.APPLICATION_JSON)
    fun hello(name: String): Publisher<Hello> = greeter.greetHello(name)
}
